src/software/backend/simulation/simulator_robot.cpp

```diff
@@ -1,382 +1,437 @@
     return 0.0;
 }
 
-float SimulatorRobotSingleton::getVelocityY()
+Point SimulatorRobot::position()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    return Point(getPositionX(), getPositionY());
+}
+
+float SimulatorRobot::getOrientation()
+{
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        return robot->orientation().toRadians();
     }
+    LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot" << std::endl;
     return 0.0;
 }
 
-float SimulatorRobotSingleton::getVelocityAngular()
+float SimulatorRobot::getVelocityX()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        return robot->velocity().x();
     }
+    LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot" << std::endl;
     return 0.0;
 }
 
-float SimulatorRobotSingleton::getBatteryVoltage()
+float SimulatorRobot::getVelocityY()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        return robot->velocity().y();
     }
+    LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot" << std::endl;
     return 0.0;
 }
 
-void SimulatorRobotSingleton::kick(float speed_m_per_s)
+Vector SimulatorRobot::velocity()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
-    {
-        // TODO: Implement me
-    }
+    return Vector(getVelocityX(), getVelocityY());
 }
 
-void SimulatorRobotSingleton::chip(float distance_m)
+float SimulatorRobot::getVelocityAngular()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        return robot->angularVelocity().toRadians();
     }
+    LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot" << std::endl;
+    return 0.0;
 }
 
-void SimulatorRobotSingleton::enableAutokick(float speed_m_per_s)
+float SimulatorRobot::getBatteryVoltage()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
-    {
-        // TODO: Implement me
-    }
+    // We currently have 4s batteries on the robot that charge up to a little over
+    // 16V, so we use 16 here to approxiamte a fully-charged battery
+    // TODO: Should max battery voltage be a constant / injected robot param?
+    return 16.0;
 }
 
-void SimulatorRobotSingleton::enableAutochip(float distance_m)
+void SimulatorRobot::kick(float speed_m_per_s)
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        if (ball_in_dribbler_area)
+        {
+            Vector kick_vector = Vector::createFromAngle(
+                robot->getRobotWithTimestamp(Timestamp::fromSeconds(0)).orientation());
+            kick_vector = kick_vector.normalize(speed_m_per_s);
+            ball_in_dribbler_area->kick(kick_vector);
+        }
```

Comment:
As a thought, are we assuming that there's only ever one ball at a time?

Reply:
Yes, in that the simulator robot only stores a single pointer to balls in its dribbler area. Otherwise the system could in theory handle multiple balls, since it just looks for a ball to enter / be in that zone.

Is this just a question, or do you think that assumption should be documented somewhere?

Reply:
It would be good to document it somehow, in some cases (like for the `SimulatorBallSingleton`), we work to allow multiple balls, but for stuff like the dribbler we only permit one. Might be worth an in-person chat about where to doc this though, I'm really not super sure.

Reply:
code now handles multiple balls